test/parallel/xyk-pallet.API.pool.weightedPools.test.ts

```diff
@@ -179,6 +181,107 @@ test("Testing that the sum of the weights can be greater than 100", async () =>
   );
 });
 
+test("GIVEN a pool WHEN it has configured with 0 THEN no new issuance will be reserved AND user can't claim rewards", async () => {
+  const [testUser2] = setupUsers();
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(token1, testUser2, Assets.DEFAULT_AMOUNT),
+    Assets.mintNative(testUser2),
+    Sudo.sudoAs(
+      testUser1,
+      Xyk.mintLiquidity(MGA_ASSET_ID, token1, defaultCurrencyValue)
+    )
+  );
+
+  testUser2.addAsset(MGA_ASSET_ID);
+  testUser2.addAsset(token1);
+
+  await waitForRewards(testUser1, liqId);
+
+  await claimRewardsAll(testUser1, liqId).then((result) => {
+    const eventResponse = getEventResultFromMangataTx(result);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+  });
+
+  await waitForRewards(testUser1, liqId);
+
+  await promotePool(sudo.keyRingPair, liqId, 0);
+
+  await claimRewardsAll(testUser1, liqId).then((result) => {
+    const eventResponse = getEventResultFromMangataTx(result);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+    expect(eventResponse.data).toEqual("NotAPromotedPool");
+  });
+
+  await mintLiquidity(
+    testUser2.keyRingPair,
+    MGA_ASSET_ID,
+    token1,
+    defaultCurrencyValue
+  );
+
+  await testUser2.refreshAmounts(AssetWallet.AFTER);
+
+  expect(testUser2.getAsset(MGA_ASSET_ID)!.amountAfter.free!).bnGt(BN_ZERO);
+  expect(testUser2.getAsset(MGA_ASSET_ID)!.amountAfter.reserved!).bnEqual(
+    BN_ZERO
+  );
+});
+
+test("GIVEN a deactivated pool WHEN its configured with more weight, THEN rewards are now active, new users can get portion of those rewards AND issuance grows", async () => {
+  const api = getApi();
+
+  await promotePool(sudo.keyRingPair, liqId, 0);
+
+  const [testUser2] = setupUsers();
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(token1, testUser2, Assets.DEFAULT_AMOUNT),
+    Assets.mintNative(testUser2)
+  );
+
+  const totalIssuanceBefore = new BN(
+    await api.query.tokens.totalIssuance(liqId)
```

Comment:
have you tried usign this? 
issuance.promotedPoolsRewardsV2()

are not the rewards growing on each pool?  IDK what is totalIssuance about :(

Reply:
The function was replaced, I also changed the async function which receives information about the pool's rewards and weight